src/SprykerEco/Zed/OauthAzure/Business/Reader/ResourceOwnerReader.php

```diff
@@ -0,0 +1,107 @@
+use SprykerEco\Zed\OauthAzure\Dependency\External\OauthAzureToOauthAdapterInterface;
+use SprykerEco\Zed\OauthAzure\OauthAzureConfig;
+use Throwable;
+
+class ResourceOwnerReader implements ResourceOwnerReaderInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @var \SprykerEco\Zed\OauthAzure\Dependency\External\OauthAzureToOauthAdapterInterface
+     */
+    protected $oauthAdapter;
+
+    /**
+     * @var \SprykerEco\Zed\OauthAzure\Business\Mapper\ResourceOwnerMapperInterface
+     */
+    protected $resourceOwnerMapper;
+
+    /**
+     * @var \SprykerEco\Zed\OauthAzure\Business\Validator\StateParameterValidatorInterface
+     */
+    protected $stateParameterValidator;
+
+    /**
+     * @param \SprykerEco\Zed\OauthAzure\Dependency\External\OauthAzureToOauthAdapterInterface $oauthAdapter
+     * @param \SprykerEco\Zed\OauthAzure\Business\Mapper\ResourceOwnerMapperInterface $resourceOwnerMapper
+     * @param \SprykerEco\Zed\OauthAzure\Business\Validator\StateParameterValidatorInterface $stateParameterValidator
+     */
+    public function __construct(
+        OauthAzureToOauthAdapterInterface $oauthAdapter,
+        ResourceOwnerMapperInterface $resourceOwnerMapper,
+        StateParameterValidatorInterface $stateParameterValidator
+    ) {
+        $this->oauthAdapter = $oauthAdapter;
+        $this->resourceOwnerMapper = $resourceOwnerMapper;
+        $this->stateParameterValidator = $stateParameterValidator;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ResourceOwnerRequestTransfer $resourceOwnerRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\ResourceOwnerResponseTransfer
+     */
+    public function getResourceOwner(
+        ResourceOwnerRequestTransfer $resourceOwnerRequestTransfer
+    ): ResourceOwnerResponseTransfer {
+        $resourceOwnerRequestTransfer
+            ->requireCode()
+            ->requireState();
+
+        $resourceOwnerResponseTransfer = (new ResourceOwnerResponseTransfer())
+            ->setIsSuccessful(true);
+
+        if (!$this->stateParameterValidator->isValidStateParameter($resourceOwnerRequestTransfer->getStateOrFail())) {
+            return $resourceOwnerResponseTransfer->setIsSuccessful(false);
+        }
+
+        return $this->fetchResourceOwner($resourceOwnerRequestTransfer, $resourceOwnerResponseTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ResourceOwnerRequestTransfer $resourceOwnerRequestTransfer
+     * @param \Generated\Shared\Transfer\ResourceOwnerResponseTransfer $resourceOwnerResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\ResourceOwnerResponseTransfer
+     */
+    protected function fetchResourceOwner(
+        ResourceOwnerRequestTransfer $resourceOwnerRequestTransfer,
+        ResourceOwnerResponseTransfer $resourceOwnerResponseTransfer
+    ): ResourceOwnerResponseTransfer {
+        try {
+            /** @var \League\OAuth2\Client\Token\AccessToken $accessToken */
```

Comment:
I have tried to change it in the adapter but it does not help because PHP stan see that return type ` \League\OAuth2\Client\Token\AccessTokenInterface` not the `\League\OAuth2\Client\Token\AccessToken`.